### PR TITLE
[Keyboard] Pink Labs e88

### PR DESCRIPTION
**Pull request by 2-n**
## Description
Adding the keymap for the Pink Labs e88 PCB, for the Bliss keyboard.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/e88-f13/rules.mk [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = NO        # Enable keyboard RGB underglow
+MIDI_ENABLE = no            # MIDI support
+UNICODE_ENABLE = no         # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output on port C6
+FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
+HD44780_ENABLE = no         # Enable support for HD44780 based LCDs
```
Comment (ridingqwerty) [inline keyboards/e88-f13/rules.mk:33]:
```suggestion
```

File: keyboards/e88-f13/readme.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# e88-f13
```
Comment (ridingqwerty) [inline keyboards/e88-f13/readme.md:1]:
```suggestion
# e88_f13
```

File: keyboards/e88-f13/readme.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+![pink labs logo](https://cdn.shopify.com/s/files/1/0302/4198/0548/files/logo_x60.png?v=1579122787)
+
+* Maintainer: [elli](https://github.com/2-n)
+* Hardware Supported: e88 f13 PCB (proto and final rev)
+* Hardware Availability: [labs.pink](http://www.labs.pink/)
+
+Make example for this keyboard (after setting up your build environment):
+
+```sh
+make e88-f13:default
```
Comment (ridingqwerty) [inline keyboards/e88-f13/readme.md:12]:
```suggestion
make e88_f13:default
```

File: keyboards/e88-f13/readme.md [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+```
+
+Example of flashing this keyboard:
+
+```sh
+make e88-f13:default:flash
```
Comment (ridingqwerty) [inline keyboards/e88-f13/readme.md:18]:
```suggestion
make e88_f13:default:flash
```

File: keyboards/e88-f13/keymaps/default/keymap.c [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+ */
+#include QMK_KEYBOARD_H
+
+// Each layer gets a name for readability, which is then used in the keymap matrix below.
+// The underscores don't mean anything - you can have a layer called STUFF or any other name.
+// Layer names don't all need to be of the same length, obviously, and you can also skip them
+// entirely and just use numbers.
+
+#define _MA 0 //Main layer
+#define _FN 1 //Main layer
```
Comment (ridingqwerty) [inline keyboards/e88-f13/keymaps/default/keymap.c:24]:
```suggestion
#define _FN 1 //Function layer

```

File: keyboards/e88-f13/config.h [hunk 6]
```diff
@@ -0,0 +1,251 @@
...
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0x4705
+#define PRODUCT_ID      0x0187
+#define DEVICE_VER      0x0001
+#define MANUFACTURER    Pink Labs
+#define PRODUCT         F13
+#define DESCRIPTION     A TKL
```
Comment (ridingqwerty) [inline keyboards/e88-f13/config.h:27]:
```suggestion
```

File: keyboards/e88-f13/config.h [hunk 7]
```diff
@@ -0,0 +1,251 @@
...
+#define LCD_DATA2_PIN    6            //< pin for 4bit data bit 2
+#define LCD_DATA3_PIN    7            //< pin for 4bit data bit 3
+#define LCD_RS_PORT      LCD_PORT     //< port for RS line
+#define LCD_RS_PIN       3            //< pin  for RS line
+#define LCD_RW_PORT      LCD_PORT     //< port for RW line
+#define LCD_RW_PIN       2            //< pin  for RW line
+#define LCD_E_PORT       LCD_PORT     //< port for Enable line
+#define LCD_E_PIN        1            //< pin  for Enable line
+#endif
+*/
```
Comment (ridingqwerty) [inline keyboards/e88-f13/config.h:244]:
```suggestion
```

**Comment by 2-n** [comment]
Fixed everything that was of issue and removed unnecessary things from config like you stated. Didn't realize files needed to be same name as folder was confused myself, but everything was renamed to prevent any further issue and now no problems with either check.

File: keyboards/e88/config.h [hunk 8]
```diff
@@ -0,0 +1,162 @@
...
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0x4705
+#define PRODUCT_ID      0x0187
+#define DEVICE_VER      0x0001
+#define MANUFACTURER    Pink Labs
+#define PRODUCT         Bliss
```
Comment (ridingqwerty) [inline keyboards/e88/config.h:26]: Why "Bliss" here and not "e88" or the like?

Comment (2-n) [reply keyboards/e88/config.h:26]: PCB was designed for my board Bliss.

File: keyboards/e88/readme.md [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+make e88:default
+```
+
+Example of flashing this keyboard:
+
+```sh
+make e88:default:flash
+```
+
+See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
```
Comment (zvecr) [inline keyboards/e88/readme.md:21]:
```suggestion
Make example for this keyboard (after setting up your build environment):

    make e88:default

Flashing example for this keyboard:

    make e88:default:flash

See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
```

File: keyboards/e88/keymaps/default/keymap.c [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+ */
+#include QMK_KEYBOARD_H
+
+// Each layer gets a name for readability, which is then used in the keymap matrix below.
+// The underscores don't mean anything - you can have a layer called STUFF or any other name.
+// Layer names don't all need to be of the same length, obviously, and you can also skip them
+// entirely and just use numbers.
+
+#define _MA 0 //Main layer
+#define _FN 1 //Function layer
```
Comment (zvecr) [inline keyboards/e88/keymaps/default/keymap.c:24]:
```suggestion
// Defines names for use in layer keycodes and the keymap
enum layer_names {
    _MA, //Main layer
    _FN  //Function layer
};
```

File: keyboards/e88/rules.mk [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+COMMAND_ENABLE = yes        # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = NO        # Enable keyboard RGB underglow
+MIDI_ENABLE = no            # MIDI support
+UNICODE_ENABLE = no         # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
```
Comment (zvecr) [inline keyboards/e88/rules.mk:30]:
```suggestion
# Bootloader selection
BOOTLOADER = qmk-dfu

# Build Options
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = no            # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
```

File: keyboards/e88/config.h [hunk 12]
```diff
@@ -0,0 +1,162 @@
...
+/* disable action features */
+//#define NO_ACTION_LAYER
+//#define NO_ACTION_TAPPING
+//#define NO_ACTION_ONESHOT
+//#define NO_ACTION_MACRO
+//#define NO_ACTION_FUNCTION
+
+/* Bootmagic Lite key configuration */
+// #define BOOTMAGIC_LITE_ROW 0
+// #define BOOTMAGIC_LITE_COLUMN 0
```
Comment (zvecr) [inline keyboards/e88/config.h:159]:
```suggestion
/* If defined, GRAVE_ESC will always act as ESC when CTRL is held.
 * This is useful for the Windows task manager shortcut (ctrl+shift+esc).
 */
//#define GRAVE_ESC_CTRL_OVERRIDE

/*
 * Force NKRO
 *
 * Force NKRO (nKey Rollover) to be enabled by default, regardless of the saved
 * state in the bootmagic EEPROM settings. (Note that NKRO must be enabled in the
 * makefile for this to work.)
 *
 * If forced on, NKRO can be disabled via magic key (default = LShift+RShift+N)
 * until the next keyboard reset.
 *
 * NKRO may prevent your keystrokes from being detected in the BIOS, but it is
 * fully operational during normal computer usage.
 *
 * For a less heavy-handed approach, enable NKRO via magic key (LShift+RShift+N)
 * or via bootmagic (hold SPACE+N while plugging in the keyboard). Once set by
 * bootmagic, NKRO mode will always be enabled until it is toggled again during a
 * power-up.
 *
 */
//#define FORCE_NKRO

/*
 * Feature disable options
 *  These options are also useful to firmware size reduction.
 */

/* disable debug print */
//#define NO_DEBUG

/* disable print */
//#define NO_PRINT

/* disable action features */
//#define NO_ACTION_LAYER
//#define NO_ACTION_TAPPING
//#define NO_ACTION_ONESHOT

/* disable these deprecated features by default */
#define NO_ACTION_MACRO
#define NO_ACTION_FUNCTION
```

File: keyboards/e88/rules.mk [hunk 13]
```diff
@@ -0,0 +1,31 @@
...
+AUDIO_ENABLE = no           # Audio output on port C6
```
Comment (noroadsleft) [inline keyboards/e88/rules.mk:31]:
```suggestion
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
```

File: keyboards/e88/rules.mk [hunk 14]
```diff
@@ -0,0 +1,31 @@
...
+MCU = atmega32u4
+
+# Bootloader selection
+#   Teensy       halfkay
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
```
Comment (noroadsleft) [inline keyboards/e88/rules.mk:11]:
```suggestion
```

**Comment by 2-n** [comment]
> 
> 
> I'm finding myself uncomfortable with the fact that the keyboard name doesn't match the USB Product string, but I'm at a loss for a suggestion here.

Ended up changing the name to e88 in the USB Product string and adding a description. 

> 
> 
> Whoops, still missing an info.json. Otherwise LGTM

I reviewed others and I did as best of a job as I could to mimic for my board, but I think I did everything correctly. If there is anything wrong I am glad to fix it.

**Comment by 2-n** [comment]
> 
> 
> LAYOUT_all is fine, but support for the tkl_ansi and tkl_iso community layouts should really be added.
> Also a info.json is missing, documented here https://docs.qmk.fm/#/reference_info_json

Not sure how I would go about doing everything for the community layouts as this is DK Saver layout PCB with an F13 key. Though in the info.json has support for configuring all possible layouts in the one layout I have.

File: keyboards/e88/config.h [hunk 15]
```diff
@@ -23,7 +23,8 @@
 #define PRODUCT_ID      0x0187
 #define DEVICE_VER      0x0001
 #define MANUFACTURER    Pink Labs
-#define PRODUCT         Bliss
+#define PRODUCT         e88
+#define DESCRIPTION     e88 PCB for Bliss F13 TKL
```
Comment (noroadsleft) [inline keyboards/e88/config.h:27]: The `DESCRIPTION` string is actually deprecated, and no longer used anywhere within QMK.

```suggestion
```

File: keyboards/e88/info.json [hunk 16]
```diff
@@ -0,0 +1,109 @@
+{
+    "keyboard_name": "e88",
+    "url": "",
+    "maintainer": "2-n",
+    "width": 17,
+    "height": 6,
+    "layouts": {
+        "LAYOUT": {
```
Comment (noroadsleft) [inline keyboards/e88/info.json:8]: File contents are fine here, but this string needs to be an exact match for the name of the layout macro to which it applies:

https://github.com/qmk/qmk_firmware/blob/a88884870160f620c1e9161d32982701de301864/keyboards/e88/e88.h#L30

```suggestion
        "LAYOUT_all": {
```

File: keyboards/e88/readme.md [hunk 17]
```diff
@@ -0,0 +1,17 @@
+# e88
+
+![pink labs logo](https://cdn.shopify.com/s/files/1/0302/4198/0548/files/logo_x60.png?v=1579122787)
+
+* Maintainer: [elli](https://github.com/2-n)
```
Comment (noroadsleft) [inline keyboards/e88/readme.md:5]: We can call out the e88/Bliss connection here.

```suggestion
A tenkeyless PCB featuring the DK Saver layout, as designed for the Bliss keyboard by Pink Labs.

* Maintainer: [elli](https://github.com/2-n)
```

**Comment by noroadsleft** [comment]
> Not sure how I would go about doing everything for the community layouts as this is DK Saver layout PCB with an F13 key. Though in the info.json has support for configuring all possible layouts in the one layout I have.

Yeah, you're fine here. None of the community layouts support an F13 key.

File: keyboards/e88/info.json [hunk 18]
```diff
@@ -0,0 +1,109 @@
+{
+    "keyboard_name": "e88",
+    "url": "",
+    "maintainer": "2-n",
+    "width": 17,
+    "height": 6,
```
Comment (noroadsleft) [inline keyboards/e88/info.json:6]:
```suggestion
    "width": 18.25,
    "height": 6.25,
```

**Comment by 2-n** [comment]
> Everything besides this looks good to me.

Sweet, thanks. Wasn't sure if it should be the same as the matrix or the physical units.

**Comment by noroadsleft** [comment]
> Sweet, thanks. Wasn't sure if it should be the same as the matrix or the physical units.

Yeah, it's supposed to be physical units.

**Comment by noroadsleft** [comment]
Thanks!

For future reference, we recommend against committing to your `master` branch as you've done here, because pull requests from modified `master` branches can make it more difficult to keep your QMK fork updated. It is highly recommended for QMK development – regardless of what is being done or where – to keep your master updated, but **NEVER** commit to it. Instead, do all your changes in a branch (branches are basically free in Git) and issue PRs from your branches when you're developing.

There are instructions on how to keep your fork updated here:

[**Best Practices: Your Fork's Master: Update Often, Commit Never**](https://docs.qmk.fm/#/newbs_git_using_your_master_branch)

[Fixing Your Branch](https://docs.qmk.fm/#/newbs_git_resynchronize_a_branch) will walk you through fixing up your `master` branch moving forward. If you need any help with this just ask.

**Pull request merged**